### PR TITLE
Homework1

### PR DESCRIPTION
**Pull request by nagyhanna**

### CODE REVIEW

File: 30411/Nagy N. Hanna/Assignment 1.txt [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+void addFirst(int code)
+{
+    if(head==NULL)
+    {
+        head=(nodeT*)malloc(sizeof(nodeT*));
+        head->code=code;
+        head->next=NULL;
+        tail=head;
+    }
+    else if(tail==NULL)
```
Comment (Alex-Cosma) [inline 30411/Nagy N. Hanna/Assignment 1.txt]: we can somehow eliminate this second if branch.
try putting ```tail=head``` in the first if branch as a last statement and remove this whole branch entirely. Does it work? why?

File: 30411/Nagy N. Hanna/Assignment 1.txt [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    nodeT *p;
+    while(head !=NULL)
+    {
+        p=head;
+        head=head->next;
+        free(p);
+    }
+    tail=NULL;
+}
+int nr_of_list()
```
Comment (Alex-Cosma) [inline 30411/Nagy N. Hanna/Assignment 1.txt:172]: length of list

**Comment by Alex-Cosma** [comment]
Code is nice and clean, good job

**Comment by nagyhanna** [comment]
It works without the second if too. And it is logic, because if head==NULL means that there is no element in the list so there  does not exist tail; and if we have an empty list and we put only one  node then the head and the tail will be exactly the same node.  
Without "tail=head" in the first if branch the function does not work properly after the calling of the function delete_all();

**Comment by Alex-Cosma** [comment]
Why did you close the pull request!? :))

**Comment by nagyhanna** [comment]
I don t know :)))

**Comment by nagyhanna** [comment]
I have to practise because I don't know where I have to write :8ball:  :))

**Comment by Alex-Cosma** [comment]
Write what exactly?

**Comment by nagyhanna** [comment]
the replies ...

**Comment by Alex-Cosma** [comment]
Oooh, don't worry about it. :smile:

**Pull request merged**